homeassistant/components/minecraft_server/__init__.py

```diff
@@ -27,7 +36,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     domain_data = hass.data.setdefault(DOMAIN, {})
 
     # Create and store server instance.
-    assert entry.unique_id
+    entry.unique_id = entry.entry_id
     unique_id = entry.unique_id
```

Comment:
Isn't the migration doing this already?

Reply:
Yes, it is, but only for people who added the integration before and are updating HA core.

Here it is set for users who add this integration after this PR got merged. But I see your point. Here it is executed everytime the integration is set up. Before it was generated during config flow only once.

I probably can only access the config entry ID after the entry was created in the config flow. Is line 64 in the config flow the place to handle it?

```
    ...
    self.async_create_entry(title=title, data=config_data)
    # Somehow get the entry ID out of the newly created config entry and set unique ID.
    ...
```

Reply:
I tried out a few things, but couldn't find a working solution. I'm not able to get the entry ID during config flow, because it looks like the config entry is created after the config flow is finished.

Am I right? If so, then I don't see any other solution.

Reply:
Config entry can have its own entry_id as unique id. 
Either you need to have something else or remove the unique id all together.

Reply:
You mean to keep the config entry unique ID as None and only work with the config entry ID?

Reply:
Looks good, thanks @gjohansson-ST. Config entry unique ID remains now unset and only the config entry ID is used as the unique identifier.